api/v1/hostprofile_webhook_test.go

```diff
@@ -0,0 +1,149 @@
+				}
+				err := validateMemoryFunction(node, function)
+				msg := errors.New("platform memory must be allocated from 4K pages.")
+				Expect(err).To(Equal(msg))
+			})
+		})
+		Context("When memory function is not platform and size is 4kb", func() {
+			It("4K pages can only be reserved for platform memory error is thrown", func() {
+				node := MemoryNodeInfo{}
+				function := MemoryFunctionInfo{
+					Function: "random func",
+					PageSize: string(PageSize4K),
+				}
+				err := validateMemoryFunction(node, function)
+				msg := errors.New("4K pages can only be reserved for platform memory.")
+				Expect(err).To(Equal(msg))
+			})
+		})
+		Context("When memory function is  platform and size is 4kb", func() {
+			It("Validation of memory function is succesfull without error", func() {
+				node := MemoryNodeInfo{}
+				function := MemoryFunctionInfo{
+					Function: memory.MemoryFunctionPlatform,
+					PageSize: string(PageSize4K),
+				}
+				err := validateMemoryFunction(node, function)
+				Expect(err).To(BeNil())
+			})
+		})
+	})
+
+	Describe("validateProcessorInfo function is tested", func() {
+		Context("When no duplicate processor entries are present", func() {
+			It("", func() {
+				obj := &HostProfile{
+					Spec: HostProfileSpec{
+						Processors: ProcessorNodeList{
+							{
+								Functions: ProcessorFunctionList{
+									{
+										Function: "function1",
+										Count:    1,
+									},
+								},
+							},
+						},
+					},
+				}
+				err := validateProcessorInfo(obj)
+				Expect(err).To(BeNil())
+			})
+		})
+
+		Context("When there are duplicate processor entries", func() {
+			It("", func() {
+				obj := &HostProfile{
+					Spec: HostProfileSpec{
+						Processors: ProcessorNodeList{
+							{
+								Functions: ProcessorFunctionList{
+									{
+										Function: "function1",
+										Count:    1,
+									},
+									{
+										Function: "function1",
+										Count:    2,
+									},
+								},
+								Node: 5,
+							},
+						},
+					},
+				}
+				msg := errors.New("duplicate processor entries are not allowed for node 5 function function1.")
+				err := validateProcessorInfo(obj)
+				Expect(err).To(Equal(msg))
+			})
+		})
+	})
+
+	Describe("validatePhysicalVolumeInfo function is tested", func() {
+		Context("When partition size is nil ", func() {
+			It("", func() {
+				obj := &PhysicalVolumeInfo{
+					Type: physicalvolumes.PVTypePartition,
+				}
+				err := validatePhysicalVolumeInfo(obj)
+				Expect(err).To(Equal(errors.New("partition specifications must include a 'size' attribute")))
+			})
+		})
+		Context("When the volume type is not partition", func() {
+			It("", func() {
+				obj := &PhysicalVolumeInfo{
+					Type: "randomType",
+				}
+				err := validatePhysicalVolumeInfo(obj)
+				Expect(err).To(BeNil())
+			})
+		})
+	})
+
+	Describe("validateMemoryInfo function is tested", func() {
+		Context("When no duplicate memory entries are present", func() {
```

Comment:
we could add test case when duplicate memory entries are present.

Reply:
yes TBD